File: 프로그래머스/2/42578. 의상/의상.py
from collections import defaultdict
from itertools import product

def solution(clothes):
    answer = 1
    
    clothes_info = defaultdict(list)
    for item, category in clothes:
        clothes_info[category].append(item)
    
    if len(clothes_info.keys()) == 1:
        for value in clothes_info.values():
            return len(value)

    
    for value in clothes_info.values():
        answer *= len(value) + 1
    
    return answer - 1
    
    

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.HashMap;
import java.util.ArrayList;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, ArrayList<String>> hasClothes = new HashMap<>();
        
        for (String[] c : clothes) {
            ArrayList<String> list = hasClothes.getOrDefault(c[1], new ArrayList<>());
            list.add(c[0]);
            hasClothes.put(c[1], list);
        }
        
        if(hasClothes.size() == 1) {
            for(String key : hasClothes.keySet()) {
                return hasClothes.get(key).size();
            }
        }
        
        for(String key: hasClothes.keySet()) {
            answer *= hasClothes.get(key).size() + 1;
        }
        
        return answer - 1;
    }
}